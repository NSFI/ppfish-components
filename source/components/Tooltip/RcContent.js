import React from 'react';
import PropTypes from 'prop-types';

export default class RcContent extends React.Component {
  static propTypes = {
    prefixCls: PropTypes.string,
    overlay: PropTypes.oneOfType([
      PropTypes.node,
      PropTypes.func,
    ]).isRequired,
    id: PropTypes.string,
    trigger: PropTypes.any,
  }

  componentDidUpdate() {
    const { trigger } = this.props;
    if (trigger) {
      trigger.forcePopupAlign();
    }
  }

  render() {
    const { overlay, prefixCls, id } = this.props;
    return (
      <div className={`${prefixCls}-inner`} id={id}>
        {typeof overlay === 'function' ? overlay() : overlay}
      </div>
    );
  }
}
