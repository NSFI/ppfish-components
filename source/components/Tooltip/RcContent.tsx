import React from 'react';
import PropTypes from 'prop-types';
import useUpdateEffect from '../../hooks/useUpdateEffect';

export interface RcContentProps {
  prefixCls?: string;
  overlay?: (() => React.ReactNode) | React.ReactNode;
  id?: string;
  // rc-trigger 未导出 Trigger 类型
  trigger: any;
}

const RcContent = (props: RcContentProps) => {
  useUpdateEffect(() => {
    const { trigger } = props;
    if (trigger) {
      trigger.forcePopupAlign();
    }
  });

  const { overlay, prefixCls, id } = props;
  return (
    <div className={`${prefixCls}-inner`} id={id}>
      {typeof overlay === 'function' ? overlay() : overlay}
    </div>
  );
};

RcContent.propTypes = {
  prefixCls: PropTypes.string,
  overlay: PropTypes.oneOfType([PropTypes.node, PropTypes.func]).isRequired,
  id: PropTypes.string,
  trigger: PropTypes.oneOfType([PropTypes.object, PropTypes.node]),
};

export default RcContent;
