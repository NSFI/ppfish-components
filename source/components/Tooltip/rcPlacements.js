const autoAdjustOverflow = {
  adjustX: 1,
  adjustY: 1,
};

const targetOffset = [0, 0];

export const rcPlacements = {
  left: {
    points: ['cr', 'cl'],
    overflow: autoAdjustOverflow,
    offset: [-4, 0],
    targetOffset,
  },
  right: {
    points: ['cl', 'cr'],
    overflow: autoAdjustOverflow,
    offset: [4, 0],
    targetOffset,
  },
  top: {
    points: ['bc', 'tc'],
    overflow: autoAdjustOverflow,
    offset: [0, -4],
    targetOffset,
  },
  bottom: {
    points: ['tc', 'bc'],
    overflow: autoAdjustOverflow,
    offset: [0, 4],
    targetOffset,
  },
  topLeft: {
    points: ['bl', 'tl'],
    overflow: autoAdjustOverflow,
    offset: [0, -4],
    targetOffset,
  },
  leftTop: {
    points: ['tr', 'tl'],
    overflow: autoAdjustOverflow,
    offset: [-4, 0],
    targetOffset,
  },
  topRight: {
    points: ['br', 'tr'],
    overflow: autoAdjustOverflow,
    offset: [0, -4],
    targetOffset,
  },
  rightTop: {
    points: ['tl', 'tr'],
    overflow: autoAdjustOverflow,
    offset: [4, 0],
    targetOffset,
  },
  bottomRight: {
    points: ['tr', 'br'],
    overflow: autoAdjustOverflow,
    offset: [0, 4],
    targetOffset,
  },
  rightBottom: {
    points: ['bl', 'br'],
    overflow: autoAdjustOverflow,
    offset: [4, 0],
    targetOffset,
  },
  bottomLeft: {
    points: ['tl', 'bl'],
    overflow: autoAdjustOverflow,
    offset: [0, 4],
    targetOffset,
  },
  leftBottom: {
    points: ['br', 'bl'],
    overflow: autoAdjustOverflow,
    offset: [-4, 0],
    targetOffset,
  },
};

export default rcPlacements;
