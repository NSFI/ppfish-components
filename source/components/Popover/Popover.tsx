import React, { forwardRef } from 'react';
import Tooltip, { AbstractTooltipProps, TooltipRef } from '../Tooltip';

export interface PopoverProps extends AbstractTooltipProps {
  title?: React.ReactNode;
  content?: React.ReactNode;
}

const Popover = forwardRef<TooltipRef, PopoverProps>(function Popover(props, ref) {
  const { title, content, ...rest } = props;

  const getOverlay = () => {
    const { prefixCls } = props;
    return (
      <div>
        {title && <div className={`${prefixCls}-title`}>{title}</div>}
        <div className={`${prefixCls}-inner-content`}>{content}</div>
      </div>
    );
  };

  return <Tooltip {...rest} ref={ref} overlay={getOverlay()} />;
});

Popover.defaultProps = {
  prefixCls: 'fishd-popover',
  placement: 'top' as const,
  transitionName: 'zoom-big',
  trigger: 'hover' as const,
  mouseEnterDelay: 0.1,
  mouseLeaveDelay: 0.1,
  overlayStyle: {},
};

export default Popover;
