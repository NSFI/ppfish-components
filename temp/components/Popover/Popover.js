var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
import * as React from 'react';
import Tooltip from '../Tooltip';
import warning from '../../utils/warning';
var Popover = /** @class */ (function (_super) {
    __extends(Popover, _super);
    function Popover() {
        var _this = _super !== null && _super.apply(this, arguments) || this;
        _this.saveTooltip = function (node) {
            _this.tooltip = node;
        };
        return _this;
    }
    Popover.prototype.getPopupDomNode = function () {
        return this.tooltip.getPopupDomNode();
    };
    Popover.prototype.getOverlay = function () {
        var _a = this.props, title = _a.title, prefixCls = _a.prefixCls, content = _a.content;
        warning(!('overlay' in this.props), 'Popover[overlay] is removed, please use Popover[content] instead, ' +
            'see: https://u.ant.design/popover-content');
        return (React.createElement("div", null,
            title && React.createElement("div", { className: prefixCls + "-title" }, title),
            React.createElement("div", { className: prefixCls + "-inner-content" }, content)));
    };
    Popover.prototype.render = function () {
        var props = __assign({}, this.props);
        delete props.title;
        return React.createElement(Tooltip, __assign({}, props, { ref: this.saveTooltip, overlay: this.getOverlay() }));
    };
    Popover.defaultProps = {
        prefixCls: 'fishd-popover',
        placement: 'top',
        transitionName: 'zoom-big',
        trigger: 'hover',
        mouseEnterDelay: 0.1,
        mouseLeaveDelay: 0.1,
        overlayStyle: {}
    };
    return Popover;
}(React.Component));
export default Popover;
