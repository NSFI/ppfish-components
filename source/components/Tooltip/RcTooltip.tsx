import * as React from 'react';
import Trigger from 'rc-trigger';
import type { TriggerProps } from 'rc-trigger';
import type { AlignType, AnimationType, ActionType } from 'rc-trigger/lib/interface';
import { rcPlacements } from './rcPlacements';
import RcContent from './RcContent';
import PropTypes from 'prop-types';

export interface TooltipProps extends Pick<TriggerProps, 'onPopupAlign' | 'builtinPlacements'> {
  trigger?: ActionType | ActionType[];
  children?: React.ReactElement;
  defaultVisible?: boolean;
  visible?: boolean;
  placement?: string;
  transitionName?: string;
  animation?: AnimationType;
  onVisibleChange?: (visible: boolean) => void;
  afterVisibleChange?: (visible: boolean) => void;
  overlay: (() => React.ReactNode) | React.ReactNode;
  overlayStyle?: React.CSSProperties;
  overlayClassName?: string;
  prefixCls?: string;
  mouseEnterDelay?: number;
  mouseLeaveDelay?: number;
  getTooltipContainer?: (node: HTMLElement) => HTMLElement;
  destroyTooltipOnHide?: boolean;
  align?: AlignType;
  arrowContent?: React.ReactNode;
  id?: string;
  popupVisible?: boolean;
  zIndex?: number;
}

export interface RcTooltipRef {
  // rc-trigger 未导出 Trigger 类型
  trigger: React.Component<TriggerProps>;
  getPopupDomNode: () => HTMLElement;
}

const RcInternalTooltip: React.ForwardRefRenderFunction<RcTooltipRef, TooltipProps> = (
  props,
  ref,
) => {
  const {
    overlayClassName,
    trigger,
    mouseEnterDelay,
    mouseLeaveDelay,
    overlayStyle,
    prefixCls,
    children,
    onVisibleChange,
    afterVisibleChange,
    transitionName,
    animation,
    placement,
    align,
    destroyTooltipOnHide,
    defaultVisible,
    getTooltipContainer,
    ...restProps
  } = props;

  const extraProps = { ...restProps };
  if ('visible' in props) {
    extraProps.popupVisible = props.visible;
  }

  const triggerRef = React.useRef(null);

  const getPopupElement = () => {
    const { arrowContent, overlay, prefixCls, id } = props;
    return [
      <div className={`${prefixCls}-arrow`} key="arrow">
        {arrowContent}
      </div>,
      <RcContent
        key="content"
        trigger={triggerRef.current}
        prefixCls={prefixCls}
        id={id}
        overlay={overlay}
      />,
    ];
  };

  const getPopupDomNode = () => {
    return triggerRef.current.getPopupDomNode();
  };

  React.useImperativeHandle(ref, () => ({
    trigger: triggerRef.current,
    getPopupDomNode,
  }));

  return (
    <Trigger
      popupClassName={overlayClassName}
      ref={triggerRef}
      prefixCls={prefixCls}
      popup={getPopupElement}
      action={trigger}
      builtinPlacements={rcPlacements}
      popupPlacement={placement}
      popupAlign={align}
      getPopupContainer={getTooltipContainer}
      onPopupVisibleChange={onVisibleChange}
      afterPopupVisibleChange={afterVisibleChange}
      popupTransitionName={transitionName}
      popupAnimation={animation}
      defaultPopupVisible={defaultVisible}
      destroyPopupOnHide={destroyTooltipOnHide}
      mouseLeaveDelay={mouseLeaveDelay}
      popupStyle={overlayStyle}
      mouseEnterDelay={mouseEnterDelay}
      {...extraProps}
    >
      {children}
    </Trigger>
  );
};

const RcTooltip = React.forwardRef(RcInternalTooltip);

RcTooltip.propTypes = {
  trigger: PropTypes.any,
  children: PropTypes.any,
  defaultVisible: PropTypes.bool,
  visible: PropTypes.bool,
  placement: PropTypes.string,
  transitionName: PropTypes.oneOfType([PropTypes.string, PropTypes.object]),
  animation: PropTypes.any,
  onVisibleChange: PropTypes.func,
  afterVisibleChange: PropTypes.func,
  overlay: PropTypes.oneOfType([PropTypes.node, PropTypes.func]).isRequired,
  overlayStyle: PropTypes.object,
  overlayClassName: PropTypes.string,
  prefixCls: PropTypes.string,
  mouseEnterDelay: PropTypes.number,
  mouseLeaveDelay: PropTypes.number,
  getTooltipContainer: PropTypes.func,
  destroyTooltipOnHide: PropTypes.bool,
  align: PropTypes.object,
  arrowContent: PropTypes.any,
  id: PropTypes.string,
};

RcTooltip.defaultProps = {
  prefixCls: 'fishd-tooltip',
  mouseEnterDelay: 0,
  destroyTooltipOnHide: false,
  mouseLeaveDelay: 0.1,
  align: {},
  placement: 'right',
  trigger: ['hover'],
  arrowContent: null,
};

export default RcTooltip;
