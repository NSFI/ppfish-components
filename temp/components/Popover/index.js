import Popover from './Popover';
import './style/index.less';
export default Popover;
