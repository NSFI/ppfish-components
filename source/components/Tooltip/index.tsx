import * as React from 'react';
import { cloneElement } from 'react';
import * as RcTooltip from './RcTooltip';
import classNames from 'classnames';
import getPlacements, { AdjustOverflow, PlacementsConfig } from './placements';
import Button from '../Button';

export { AdjustOverflow, PlacementsConfig };

export type TooltipPlacement =
  'top' | 'left' | 'right' | 'bottom' |
  'topLeft' | 'topRight' | 'bottomLeft' | 'bottomRight' |
  'leftTop' | 'leftBottom' | 'rightTop' | 'rightBottom';

export type TooltipTrigger = 'hover' | 'focus' | 'click' | 'contextMenu';

export interface AbstractTooltipProps {
  prefixCls?: string;
  overlayClassName?: string;
  style?: React.CSSProperties;
  overlayStyle?: React.CSSProperties;
  placement?: TooltipPlacement;
  builtinPlacements?: Object;
  defaultVisible?: boolean;
  visible?: boolean;
  onVisibleChange?: (visible: boolean) => void;
  mouseEnterDelay?: number;
  mouseLeaveDelay?: number;
  transitionName?: string;
  trigger?: TooltipTrigger;
  openClassName?: string;
  arrowPointAtCenter?: boolean;
  autoAdjustOverflow?: boolean | AdjustOverflow;
  // getTooltipContainer had been rename to getPopupContainer
  getTooltipContainer?: (triggerNode: Element) => HTMLElement;
  getPopupContainer?: (triggerNode: Element) => HTMLElement;
  children?: React.ReactNode;
}

export type RenderFunction = () => React.ReactNode;

export interface TooltipProps extends AbstractTooltipProps {
  title?: React.ReactNode | RenderFunction;
  overlay?: React.ReactNode | RenderFunction;
}

const splitObject = (obj: any, keys: string[]) => {
  const picked: any = {};
  const omitted: any = { ...obj };
  keys.forEach(key => {
    if (obj && key in obj) {
      picked[key] = obj[key];
      delete omitted[key];
    }
  });
  return { picked, omitted };
};

export default class Tooltip extends React.Component<TooltipProps, any> {
  static defaultProps = {
    prefixCls: 'ant-tooltip',
    placement: 'top' as TooltipPlacement,
    transitionName: 'zoom-big-fast',
    mouseEnterDelay: 0.1,
    mouseLeaveDelay: 0.1,
    arrowPointAtCenter: false,
    autoAdjustOverflow: true,
  };

  private tooltip: typeof RcTooltip;

  constructor(props: TooltipProps) {
    super(props);

    this.state = {
      visible: !!props.visible || !!props.defaultVisible,
    };
  }

  componentWillReceiveProps(nextProps: TooltipProps) {
    if ('visible' in nextProps) {
      this.setState({ visible: nextProps.visible });
    }
  }

  onVisibleChange = (visible: boolean) => {
    const { onVisibleChange } = this.props;
    if (!('visible' in this.props)) {
      this.setState({ visible: this.isNoTitle() ? false : visible });
    }
    if (onVisibleChange && !this.isNoTitle()) {
      onVisibleChange(visible);
    }
  }

  getPopupDomNode() {
    return this.tooltip.getPopupDomNode();
  }

  getPlacements() {
    const { builtinPlacements, arrowPointAtCenter, autoAdjustOverflow } = this.props;
    return builtinPlacements || getPlacements({
      arrowPointAtCenter,
      verticalArrowShift: 8,
      autoAdjustOverflow,
    });
  }

  isHoverTrigger() {
    const { trigger } = this.props;
    if (!trigger || trigger === 'hover') {
      return true;
    }
    if (Array.isArray(trigger)) {
      return trigger.indexOf('hover') >= 0;
    }
    return false;
  }

  // Fix Tooltip won't hide at disabled button
  // mouse events don't trigger at disabled button in Chrome
  // https://github.com/react-component/tooltip/issues/18
  getDisabledCompatibleChildren(element: React.ReactElement<any>) {
    if (((element.type as typeof Button).__ANT_BUTTON || element.type === 'button') &&
        element.props.disabled && this.isHoverTrigger()) {
      // Pick some layout related style properties up to span
      // Prevent layout bugs like https://github.com/ant-design/ant-design/issues/5254
      const { picked, omitted } = splitObject(
        element.props.style,
        ['position', 'left', 'right', 'top', 'bottom', 'float', 'display', 'zIndex'],
      );
      const spanStyle = {
        display: 'inline-block',  // default inline-block is important
        ...picked,
        cursor: 'not-allowed',
      };
      const buttonStyle = {
        ...omitted,
        pointerEvents: 'none',
      };
      const child = cloneElement(element, {
        style: buttonStyle,
        className: null,
      });
      return (
        <span style={spanStyle} className={element.props.className}>
          {child}
        </span>
      );
    }
    return element;
  }

  isNoTitle() {
    const { title, overlay } = this.props;
    return !title && !overlay;  // overlay for old version compatibility
  }

  // 动态设置动画点
  onPopupAlign = (domNode: HTMLElement, align: any) => {
    const placements: any = this.getPlacements();
    // 当前返回的位置
    const placement = Object.keys(placements).filter(
      key => (
        placements[key].points[0] === align.points[0] &&
        placements[key].points[1] === align.points[1]
      ),
    )[0];
    if (!placement) {
      return;
    }
    // 根据当前坐标设置动画点
    const rect = domNode.getBoundingClientRect();
    const transformOrigin = {
      top: '50%',
      left: '50%',
    };
    if (placement.indexOf('top') >= 0 || placement.indexOf('Bottom') >= 0) {
      transformOrigin.top = `${rect.height - align.offset[1]}px`;
    } else if (placement.indexOf('Top') >= 0 || placement.indexOf('bottom') >= 0) {
      transformOrigin.top = `${-align.offset[1]}px`;
    }
    if (placement.indexOf('left') >= 0 || placement.indexOf('Right') >= 0) {
      transformOrigin.left = `${rect.width - align.offset[0]}px`;
    } else if (placement.indexOf('right') >= 0 || placement.indexOf('Left') >= 0) {
      transformOrigin.left = `${-align.offset[0]}px`;
    }
    domNode.style.transformOrigin = `${transformOrigin.left} ${transformOrigin.top}`;
  }

  saveTooltip = (node: typeof RcTooltip) => {
    this.tooltip = node;
  }

  render() {
    const { props, state } = this;
    const { prefixCls, title, overlay, openClassName, getPopupContainer, getTooltipContainer } = props;
    const children = props.children as React.ReactElement<any>;
    let visible = state.visible;
    // Hide tooltip when there is no title
    if (!('visible' in props) && this.isNoTitle()) {
      visible = false;
    }

    const child = this.getDisabledCompatibleChildren(
      React.isValidElement(children) ? children : <span>{children}</span>,
    );
    const childProps = child.props;
    const childCls = classNames(childProps.className, {
      [openClassName || `${prefixCls}-open`]: true,
    });

    return (
      <RcTooltip
        {...this.props}
        getTooltipContainer={getPopupContainer || getTooltipContainer}
        ref={this.saveTooltip}
        builtinPlacements={this.getPlacements()}
        overlay={overlay || title || ''}
        visible={visible}
        onVisibleChange={this.onVisibleChange}
        onPopupAlign={this.onPopupAlign}
      >
        {visible ? cloneElement(child, { className: childCls }) : child}
      </RcTooltip>
    );
  }
}
