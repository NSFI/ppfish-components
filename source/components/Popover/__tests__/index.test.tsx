import React from 'react';
import { mount } from 'enzyme';
import Popover from '../index';
import { TooltipRef } from '../../Tooltip';

describe('Popover', () => {
  it('should show overlay when trigger is clicked', () => {
    let popoverRef: React.RefObject<TooltipRef> = React.createRef();
    const popover = mount(
      <Popover ref={popoverRef} content="console.log('hello world')" title="code" trigger="click">
        <span>show me your code</span>
      </Popover>,
    );

    expect(popoverRef.current.getPopupDomNode()).toBe(null);

    popover.find('span').simulate('click');

    const popup = popoverRef.current.getPopupDomNode();
    expect(popup).not.toBe(null);
    expect((popover.find('Trigger PopupInner').props() as any).visible).toBeTruthy();

    expect(popup.innerHTML).toMatchSnapshot();
  });

  it('shows content for render functions', () => {
    const renderTitle = 'some-title';
    const renderContent = 'some-content';
    let popoverRef: React.RefObject<TooltipRef> = React.createRef();

    const popover = mount(
      <Popover ref={popoverRef} content={renderContent} title={renderTitle} trigger="click">
        <span>show me your code</span>
      </Popover>,
    );

    popover.find('span').simulate('click');

    const popup = popoverRef.current.getPopupDomNode();
    expect(popup).not.toBe(null);
    expect(popup.innerHTML).toContain('some-title');
    expect(popup.innerHTML).toContain('some-content');
    expect(popup.innerHTML).toMatchSnapshot();
  });

  it('handles empty title/content props safely', () => {
    let popoverRef: React.RefObject<TooltipRef> = React.createRef();

    const popover = mount(
      <Popover trigger="click" ref={popoverRef}>
        <span>show me your code</span>
      </Popover>,
    );

    popover.find('span').simulate('click');

    const popup = popoverRef.current.getPopupDomNode();
    expect(popup).not.toBe(null);
    expect(popup.innerHTML).toMatchSnapshot();
  });

  it('props#overlay do not warn anymore', () => {
    const errorSpy = jest.spyOn(console, 'error').mockImplementation(() => {});

    const overlay = jest.fn();

    // Popover 因为没有overlay类型, 想要检测只能先注释
    mount(
      // @ts-ignore
      <Popover overlay={overlay} content="console.log('hello world')" title="code" trigger="click">
        <span>show me your code</span>
      </Popover>,
    );

    expect(errorSpy.mock.calls.length).toBe(0);
    expect(overlay).not.toHaveBeenCalled();
  });
});
