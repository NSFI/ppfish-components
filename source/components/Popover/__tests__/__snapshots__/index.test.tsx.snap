// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Popover handles empty title/content props safely 1`] = `"<div class=\\"fishd-popover-content\\"><div class=\\"fishd-popover-arrow\\"></div><div class=\\"fishd-popover-inner\\"><div><div class=\\"fishd-popover-inner-content\\"></div></div></div></div>"`;

exports[`Popover should show overlay when trigger is clicked 1`] = `"<div class=\\"fishd-popover-content\\"><div class=\\"fishd-popover-arrow\\"></div><div class=\\"fishd-popover-inner\\"><div><div class=\\"fishd-popover-title\\">code</div><div class=\\"fishd-popover-inner-content\\">console.log('hello world')</div></div></div></div>"`;

exports[`Popover shows content for render functions 1`] = `"<div class=\\"fishd-popover-content\\"><div class=\\"fishd-popover-arrow\\"></div><div class=\\"fishd-popover-inner\\"><div><div class=\\"fishd-popover-title\\">some-title</div><div class=\\"fishd-popover-inner-content\\">some-content</div></div></div></div>"`;
